### PR TITLE
📚 Add docstrings to parts of myst-cli

### PR DESCRIPTION
**Pull request by agoose77**
This PR changes no functionality; I wanted to familiarise myself with some of the inner workings through refactoring. There are some new docstrings that mostly state the obvious but it's a start.

### CODE REVIEW

File: packages/myst-cli/src/build/jats/single.ts [hunk 1]
```diff
@@ -29,6 +35,9 @@ export async function runJatsExport(
   const toc = tic();
   const { output, articles, sub_articles } = exportOptions;
   // At this point, export options are resolved to contain one-and-only-one article
+  if (articles.length > 1) {
+    throw new Error('When specifying a named output for export, you must list exactly one file.');
+  }
```
Comment (agoose77) [inline packages/myst-cli/src/build/jats/single.ts:40]: The verbal asserts imply `articles.length === 1`, but the later code permits `.length === 0`, so I've imposed this.

File: packages/myst-cli/src/build/md/index.ts [hunk 2]
```diff
@@ -27,6 +27,9 @@ export async function runMdExport(
   const toc = tic();
   const { output, articles } = exportOptions;
   // At this point, export options are resolved to contain one-and-only-one article
+  if (articles.length > 1) {
+    throw new Error('When specifying a named output for export, you must list exactly one file.');
+  }
```
Comment (agoose77) [inline packages/myst-cli/src/build/md/index.ts:32]: The verbal asserts imply `articles.length === 1`, but the later code permits `.length === 0`, so I've imposed this.

File: packages/myst-cli/src/process/site.ts [hunk 3]
```diff
@@ -107,20 +125,6 @@ export async function addProjectReferencesToObjectsInv(
...
-) {
-  const project = await loadProjectFromDisk(session, projectPath, {
-    warnOnNoConfig: true,
-    ...opts,
-  });
-  // Load the citations first, or else they are loaded in each call below
-  const pages = filterPages(project);
-  return { project, pages };
-}
-
```
Comment (agoose77) [inline packages/myst-cli/src/process/site.ts:123]: This function is very "thin", so I decided to remove it for readability. Can restore if needed!

File: packages/myst-cli/src/process/mdast.ts [hunk 4]
```diff
@@ -79,16 +79,6 @@ const LINKS_SELECTOR = 'link,card,linkBlock';
...
-const htmlHandlers = {
-  comment(h: any, node: any) {
-    // Prevents HTML comments from showing up as text in web
-    // TODO: Remove once this is landed in myst-parser
-    const result = h(node, 'comment');
-    (result as any).value = node.value;
-    return result;
-  },
-};
-
```
Comment (agoose77) [inline packages/myst-cli/src/process/mdast.ts:91]: It looks like myst-transforms does this now.

**Comment by rowanc1** [comment]
I would love to get these changes in. I think that it would be helpful to break them up into independent PRs I think though, which would really help in reviewing and merging speed.

Potentially like 3-4 PRs - one on doc strings, one on the loading refactor, removing the HTML comments handler, and one on limiting the articles length. I can help with the cherry-picking/rebasing if you want.

We can talk more on monday!

**Comment by agoose77** [comment]
@rowanc1 this PR is now *just* docstrings! :)

**Review by fwkoch — APPROVED** [review]
This is great, thanks!

**Comment by rowanc1** [comment]
@agoose77 -- just for clarity, the other changes ideally can come in as well, just broken up so they can be reviewed independently! Hoping you have some of the changes locally. :)

Thanks for your help in cleaning these up!

**Pull request merged**